crates/router/src/db/kafka_store.rs

```diff
@@ -165,7 +167,7 @@ impl AddressInterface for KafkaStore {
     async fn find_address_by_merchant_id_payment_id_address_id(
         &self,
         state: &KeyManagerState,
-        merchant_id: &id_type::MerchantId,
+        merchant_id: &MerchantId,
```

Comment:
Hey @qstommyshu I think we could have qualified here and removed the import. Since all other types are `id_types::ID`, like PaymentsId, CustomerId etc.

Reply:
Hi @apoorvdixit88 ,
I don't think we can qualify `MerchantId`, the qualification here is exactly the reason why some of the workflow failed many times (just like how it looks like in my previous screenshot). The only way to resolve those errors is to remove the qualification and use `import`.

Reply:
All are `id_types`. We can try removing import `MerchantId` and then qualify, and see if it works. I think the error was because of feature flag only. All the checks are not causing issue to other PRs in which merchant_id is qualified everywhere.

Reply:
I've tried this approach, in [ae5b627](https://github.com/juspay/hyperswitch/pull/6293/commits/ae5b62798a53362d1f0dd65b423c494a1bf2930f), I only added the feature flags to see if the error message in my local console disappears, but it did not. Then I tried replacing those qualification in the next commit [e886431](https://github.com/juspay/hyperswitch/pull/6293/commits/e886431681ce8a5e1b9e306760eb8c362e34f264) to what `cargo clippy` suggests, and everything is cleared since then.

Reply:
> Hi @apoorvdixit88 , I don't think we can qualify `MerchantId`, the qualification here is exactly the reason why some of the workflow failed many times (just like how it looks like in my previous screenshot). [...]

Hey @qstommyshu, I think `cargo clippy` tells about unused qualification because of the `id_type::MerchantId` import added towards the beginning of the file. Can you remove that import, then qualify all uses of `MerchantId` to be `id_type::MerchantId`?

Reply:
> > Hi @apoorvdixit88 , I don't think we can qualify `MerchantId`, the qualification here is exactly the reason why some of the workflow failed many times (just like how it looks like in my previous screenshot). [...]
> 
> Hey @qstommyshu, I think `cargo clippy` tells about unused qualification because of the `id_type::MerchantId` import added towards the beginning of the file. Can you remove that import, then qualify all uses of `MerchantId` to be `id_type::MerchantId`?
> 
> We'd ideally want to avoid unrelated changes to ease PR reviews.

Hi @SanchithHegde ,

Thanks for the reply, I've just checked your suggestion is correct,  and I just update the code as your suggestion. Please review the update when you are available.

And for @apoorvdixit88 , sorry for the wrong information, what you thought about is correct. The error is not caused by `MerchantId`.